build_tools/github_actions/test_executable_scripts/test_hiptests.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import os
+import shlex
+import subprocess
+from pathlib import Path
+import glob
+import shutil
+import json
+import sys
+import platform
+
+logging.basicConfig(level=logging.INFO)
+THEROCK_BIN_DIR_STR = os.getenv("THEROCK_BIN_DIR")
+if THEROCK_BIN_DIR_STR is None:
+    logging.info("++ Error: env(THEROCK_BIN_DIR) is not set. Please set it before executing tests.")
+    sys.exit(1)
+THEROCK_BIN_DIR = Path(THEROCK_BIN_DIR_STR)
+SCRIPT_DIR = Path(__file__).resolve().parent
+THEROCK_DIR = SCRIPT_DIR.parent.parent.parent
+
+SHARD_INDEX = os.getenv("SHARD_INDEX", 1)
+TOTAL_SHARDS = os.getenv("TOTAL_SHARDS", 1)
+CATCH_TESTS_PATH = f"{THEROCK_BIN_DIR}/../share/hip/catch_tests"
+if not os.path.isdir(CATCH_TESTS_PATH):
+    logging.info(f"++ Error: catch tests not found in {CATCH_TESTS_PATH}")
+    sys.exit(1)
+
+env = os.environ.copy()
+
+def get_test_count():
+    cmd = ["ctest", "--show-only=json-v1"]
+    result = subprocess.run(
+        cmd,
+        cwd=CATCH_TESTS_PATH,
+        check=True,
+        capture_output=True,
+    )
+    jdata = json.loads(result.stdout)
+    tests = jdata["tests"]
+    return len(tests)
+
+
+def get_test_range_per_shard(total_test_count: int, total_shards, shard_index):
+    tests_per_shard = int(total_test_count / total_shards)
+    current_index = (tests_per_shard * (shard_index - 1)) + 1
+    end_index = current_index + tests_per_shard - 1
+    if shard_index == total_shards:
+        # Retrieve remaining tests
+        end_index = total_test_count
+    logging.info(
+        f"""++ hip-tests ctest: shard {shard_index} / {total_shards}. Running:{tests_per_shard} tests""")
+    return [current_index, end_index]
+
+
+if platform.system() == "Windows":
```

Comment:
it seems like this indentation is wrong? should this be part of `get_test_range_per_shard()`?
if not please put everything in a function and guard it by `if __name__ == "__main__":`